Library/Homebrew/os/mac.rb

```diff
@@ -37,6 +37,15 @@ def outdated_release?
       version < "10.9"
     end
 
+    def os_name
```

Comment:
You'll need to actually pick up the changes I suggested [here](https://github.com/Homebrew/brew/issues/961#issuecomment-247051002). Plain version comparisons like this don't work especially well, you get situations where `10.10` is correctly reported as `OS X` but `10.7` is flagged as `macOS`.

The proposed change in the linked issue also allows you to fix the situation where Requirements have their own `@version` we shouldn't be ignoring. My proposed fix in the other issue isn't perfect, and will need some additional testing & tweaking, but it's closer.